src/transformers/models/seamless_m4t/tokenization_seamless_m4t.py

```diff
@@ -0,0 +1,612 @@
+            token=token,
+            cache_dir=cache_dir,
+            local_files_only=local_files_only,
+            _commit_hash=_commit_hash,
+            _is_local=_is_local,
+            **kwargs,
+        )
+
+        # needs to recompute after loading from pretrained
+        # Mimic fairseq token-to-id alignment for the first 4 token
+
+        tokenizer.fairseq_tokens_to_ids = {"<pad>": 0, "<unk>": 1, "<s>": 2, "</s>": 3}
+
+        language_code = [
+            tok for tok in tokenizer.additional_special_tokens if (tok.startswith("__") and tok.endswith("__"))
+        ]
+
+        # update languages codes
+        tokenizer.lang_code_to_id = {
+            code: tokenizer.sp_model_size + i + tokenizer.fairseq_offset for i, code in enumerate(language_code)
+        }
+
+        tokenizer.id_to_lang_code = {v: k for k, v in tokenizer.lang_code_to_id.items()}
+        tokenizer.fairseq_tokens_to_ids.update(tokenizer.lang_code_to_id)
+
+        current_id = len(tokenizer.sp_model) + len(tokenizer.lang_code_to_id) + tokenizer.fairseq_offset
+        tokenizer.fairseq_tokens_to_ids["<MINED_DATA>"] = current_id
+        tokenizer.fairseq_tokens_to_ids["<MMT_BT_DATA>"] = current_id + 1
+        tokenizer.fairseq_tokens_to_ids["<SMT_BT_DATA>"] = current_id + 2
+
+        tokenizer.fairseq_ids_to_tokens = {v: k for k, v in tokenizer.fairseq_tokens_to_ids.items()}
+
+        tokenizer.src_lang = tokenizer._src_lang
+        tokenizer.tgt_lang = tokenizer._tgt_lang
+        return tokenizer
+
+    # Copied from transformers.models.nllb.tokenization_nllb.NllbTokenizer.__getstate__
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state["sp_model"] = None
+        state["sp_model_proto"] = self.sp_model.serialized_model_proto()
+        return state
+
+    # Copied from transformers.models.nllb.tokenization_nllb.NllbTokenizer.__setstate__
+    def __setstate__(self, d):
+        self.__dict__ = d
+
+        # for backward compatibility
+        if not hasattr(self, "sp_model_kwargs"):
+            self.sp_model_kwargs = {}
+
+        self.sp_model = spm.SentencePieceProcessor(**self.sp_model_kwargs)
+        self.sp_model.LoadFromSerializedProto(self.sp_model_proto)
+
+    @property
+    def vocab_size(self):
+        return (
+            len(self.sp_model) + len(self.fairseq_tokens_to_ids) - 3
+        )  # 3 for <unk>, <s> and </s> already in sp_model
+
+    def add_special_tokens(self, special_tokens_dict, replace_additional_special_tokens=True) -> int:
+        if replace_additional_special_tokens:
+            logger.warning_once(
+                "`replace_additional_special_tokens=True` will break the language token ids once saved and reloaded. Be careful with this operation."
+            )
+        return super().add_special_tokens(
+            special_tokens_dict=special_tokens_dict,
+            replace_additional_special_tokens=replace_additional_special_tokens,
+        )
+
+    def __call__(
```

Comment:
It's simply to avoid the struggle of having to reload the tokenizer every time the `tgt_lang` or `src_lang` changes!

According to its [doc](https://huggingface.co/docs/transformers/model_doc/nllb), NLLB doesn't have that flexibility:

> English (eng_Latn) is set as the default language from which to translate. In order to specify that you’d like to translate from a different language, you should specify the BCP-47 code in the src_lang keyword argument of the tokenizer initialization.

Here the user can simply seamlessly specify and change the target and source languages at every tokenizer call  without having the unnecessary burden to reload it

Reply:
Alright sounds good to me since it makes using the tokenizer easier!